### PR TITLE
Refactor embedder

### PR DESCRIPTION
**Pull request by CaptainVee**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,6 +13,9 @@ requires-python = ">=3.8"
 dependencies = [
     'sacremoses==0.0.53',
     'sentencepiece==0.1.99',
```
Comment (avidale) [inline pyproject.toml:15]: Do we really need such precise requirements? Maybe, specifying a major version of packages would be enough?

Comment (CaptainVee) [reply pyproject.toml:15]: I don't really know. I had issues with numpy when i upgraded it to use the latest version. it needed a higher version of python (python3.9 and above).

File: laser_encoders/test_laser_tokenizer.py [hunk 2]
```diff
@@ -124,3 +126,60 @@ def test_tokenize_file_overwrite(tokenizer, input_text: str):
...
+                -6.035178527235984802e-02,
+                2.210616767406463623e-01,
+                -2.701394855976104736e-01,
+                -4.902199506759643555e-01,
+                -3.126966953277587891e-02,
+            ],
+        ),
+    ],
+)
+def test_sentence_encoder(tokenizer, model_url, expected_array, input_text: str):
```
Comment (heffernankevin) [inline laser_encoders/test_laser_tokenizer.py:166]: looks like you only have type hints for the last arg?

**Review by heffernankevin — CHANGES_REQUESTED** [review]

File: source/embed.py [hunk 3]
```diff
@@ -63,129 +55,10 @@ def buffered_read(fp, buffer_size):
...
-        indices = []
-        results = []
-        for batch, batch_indices in self._make_batches(sentences):
-            indices.extend(batch_indices)
-            results.append(self._process_batch(batch))
-        return np.vstack(results)[np.argsort(indices, kind=self.sort_kind)]
-
-
-class HuggingFaceEncoder():
+class HuggingFaceEncoder:
```
[the remark this answers is not in the document]
Comment (CaptainVee) [reply source/embed.py:58]: I get a ModuleNotFoundError: No module named 'laser_encoders'. when I run the script.

[the remark this answers is not in the document]
Comment (heffernankevin) [reply source/embed.py:58]: It's not ideal, but one option here which doesn't break existing functionality (and which I've seen in other files such as `text_processing.py`) is to add this to `embed.py`:

```
assert os.environ.get("LASER"), "Please set the environment variable LASER"
LASER = os.environ["LASER"]
sys.path.append(LASER)
```

[the remark this answers is not in the document]
Comment (CaptainVee) [reply source/embed.py:58]: It still isn't working for me. Maybe from my system but I am still getting a ModuleNotFoundError: No module named 'laser_encoders'. You should maybe try it from your end.

[the remark this answers is not in the document]
Comment (heffernankevin) [reply source/embed.py:58]: Yes it's working on my end. Did you put those lines I suggested above before you call:
`from laser_encoders.models import SentenceEncoder` in `embed.py`?

[the remark this answers is not in the document]
Comment (heffernankevin) [reply source/embed.py:58]: (also maybe just double-check as well that your env var `LASER` points to your forked repo)

**Review by avidale — APPROVED** [review]

**Pull request merged**